src/components/slideshow/react/SlideshowControls.tsx

```diff
@@ -0,0 +1,308 @@
+                        [`${CLASSNAME}__pagination-item--is-out-range`]: isPaginationItemOutVisibleRange(i),
+                    })}
+                    key={i}
+                    // tslint:disable-next-line: jsx-no-lambda
+                    onClick={(): void => handleItemClick(i)}
+                    tabIndex={-1}
+                />,
+            );
+        }
+
+        return items;
+    };
+
+    /**
+     * Handle click on an item to go to a specific slide.
+     *
+     * @param {number} index Index of the slide to go to.
+     */
+    const handleItemClick: (index: number) => void = useCallback(
+        (index: number) => {
+            if (isFunction(onPaginationClick)) {
+                onPaginationClick(index);
+            }
+        },
+        [onPaginationClick],
+    );
+
+    /**
+     * Handle click to go to next slide.
+     */
+    const handleNextClick: () => void = useCallback(() => {
+        if (isFunction(onNextClick)) {
+            onNextClick();
+        }
+    }, [onNextClick]);
+
+    /**
+     * Handle click to go to previous slide.
+     */
+    const handlePreviousClick: () => void = useCallback(() => {
+        if (isFunction(onPreviousClick)) {
+            onPreviousClick();
+        }
+    }, [onPreviousClick]);
+
+    /**
+     * Determine if a navigation item is visible.
+     *
+     * @param {number} index Index of navigation item.
+     * @return {boolean} Whether navigation item is visble or not.
+     */
+    const isPaginationItemOutVisibleRange: (index: number) => boolean = (index: number): boolean => {
+        return index < visibleRange.minRange || index > visibleRange.maxRange;
+    };
+
+    /**
+     * Check if the pagination item is on edge, indicating other slides after or before.
+     *
+     * @param  {number}  index The index of the pagination item to check.
+     * @return {boolean} Whether the pagination item is on edge or not.
+     */
+    const isPaginationItemOnEdge: (index: number) => boolean = (index: number): boolean => {
+        return (
+            index !== 0 && index !== lastSlide && (index === visibleRange.minRange || index === visibleRange.maxRange)
+        );
+    };
+
+    //////////////////////
+
+    const lastSlide: number = slidesCount - 1;
+    const visibleRange: PaginationRange = getVisibleRange(activeIndex);
+    const paginationItems: JSX.Element[] = buildItemsArray(lastSlide);
+
+    /**
+     * Inline style of wrapper element.
+     */
+    const wrapperStyle: {} = {
+        transform: `translateX(-${PAGINATION_ITEM_SIZE * visibleRange.minRange}px)`,
+    };
+
+    useEffect(() => {
+        if (parentRef && parentRef.current) {
+            parentRef.current.addEventListener('keydown', handleKeyPressed);
+        }
+
+        return (): void => {
+            if (parentRef && parentRef.current) {
+                parentRef.current.removeEventListener('keydown', handleKeyPressed);
+            }
+        };
+    });
```

Comment:
Maybe adding an empty array as dependency to avoid unnecessary retrigger ?
Or at least `parentRef.current` in it if `parentRef` may change